### PR TITLE
Refactor tabbable container

### PR DESCRIPTION
**Pull request by UselessToucan**
Related to https://github.com/ppy/osu-framework/issues/2423

### CODE REVIEW

**Comment by nyquillerium** [comment]
Curiously, are you working on the rest of the implementation for https://github.com/ppy/osu-framework/issues/2423? I started working drop-downs yesterday but you might be further along than I am.

File: osu.Framework/Graphics/Containers/TabbableContainer.cs [hunk 1]
```diff
@@ -36,19 +37,30 @@ public class TabbableContainer<T> : Container<T>, ITabbableContainer
...
+                    containingInputManager = GetContainingInputManager();
+
+                containingInputManager.ChangeFocus(nextTab);
+            }
+
             return true;
         }
 
-        private Drawable nextTabStop(CompositeDrawable target, bool reverse)
+        private readonly Stack<Drawable> stack = new Stack<Drawable>();
```
Comment (nyquillerium) [inline osu.Framework/Graphics/Containers/TabbableContainer.cs]: Is there a reason why this is no longer a local variable?

Comment (UselessToucan) [reply osu.Framework/Graphics/Containers/TabbableContainer.cs]: I want to reuse it in a multiple `getNextTabStop` runs. Similar to input queues in the `InputManager`

**Comment by UselessToucan** [comment]
> Curiously, are you working on the rest of the implementation for #2423?

Yes, but I am stuck on setting `TabbableContentContainer` :(



> I started working drop-downs

What exactly are you doing? Note that some of the input functionality is already implemented [here](https://github.com/ppy/osu-framework/pull/1745).

**Comment by nyquillerium** [comment]
Yeah, I noticed. I'm not sure the tab input should be handled on that level though? Might be possible to remain isolated in Tabbable container, but it depends on what we expect the behavior to be.

File: osu.Framework/Graphics/Containers/TabbableContainer.cs (hunk 1 above)
Comment (nyquillerium) [reply osu.Framework/Graphics/Containers/TabbableContainer.cs]: Guess this is alright since it has precedence

**Comment by UselessToucan** [comment]
> I'm not sure the tab input should be handled on that level though?

Of course I won't handle Tab in the dropdown itself. That PR was mentioned in order to note that we already have some of the functionality for interacting with the dropdown using keyboard.

**Comment by peppy** [comment]
Both of these feel like far too early optimisations for methods which are called very seldomly.

If this was called every frame these would have a place, but I don't think we need them otherwise.

**Comment by UselessToucan** [comment]
It does not make things worse(for example, in terms of code difficulty) so I don't see any problems with it.

**Comment by peppy** [comment]
That's not really valid reasoning. Have discussed with @smoogipoo and we both agree this is a step int he wrong direction (towards micro-optimisation).

**Pull request closed without merging**